src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -0,0 +1,455 @@
+        repositoryName = Objects.toString(repositoryName, "");
+        mirrorName = Objects.toString(mirrorName, "");
+        BitbucketRepository repository =
+                new BitbucketRepository(-1, repositoryName, null, new BitbucketProject(projectName, null, projectName),
+                        repositoryName, AVAILABLE);
+        setRepositoryDetails(credentialsId, serverId, mirrorName, repository);
+    }
+
+    private void setRepositoryDetails(@CheckForNull String credentialsId, @Nullable String serverId, String mirrorName,
+                                      BitbucketRepository repository) {
+        String cloneUrl = getCloneUrl(repository.getCloneUrls());
+        if (cloneUrl.isEmpty()) {
+            LOGGER.info("No clone url found for repository: " + repository.getName());
+        }
+        BitbucketSCMRepository bitbucketSCMRepository =
+                new BitbucketSCMRepository(credentialsId, repository.getProject().getName(),
+                        repository.getProject().getKey(), repository.getName(), repository.getSlug(),
+                        serverId, mirrorName);
+        initialize(cloneUrl, bitbucketSCMRepository, serverId);
+    }
+
+    @SuppressWarnings("Duplicates")
+    private void setRepositoryDetails(@CheckForNull String credentialsId, @Nullable String serverId,
+                                      EnrichedBitbucketMirroredRepository repository) {
+        if (isBlank(serverId)) {
+            return;
+        }
+        String cloneUrl = getCloneUrl(repository.getMirroringDetails().getCloneUrls());
+        if (cloneUrl.isEmpty()) {
+            LOGGER.info("No clone url found for repository: " + repository.getRepository().getName());
+        }
+        BitbucketRepository underlyingRepo = repository.getRepository();
+        BitbucketSCMRepository bitbucketSCMRepository =
+                new BitbucketSCMRepository(credentialsId, underlyingRepo.getProject().getName(),
+                        underlyingRepo.getProject().getKey(), underlyingRepo.getName(), underlyingRepo.getSlug(),
+                        serverId, repository.getMirroringDetails().getMirrorName());
+        initialize(cloneUrl, bitbucketSCMRepository, serverId);
+    }
+
+    @Symbol("BbS")
+    @Extension
+    @SuppressWarnings({"unused"})
+    public static class DescriptorImpl extends SCMSourceDescriptor implements BitbucketScmFormValidation,
+            BitbucketScmFormFill {
+
+        private final GitSCMSource.DescriptorImpl gitScmSourceDescriptor;
+        @Inject
+        private BitbucketClientFactoryProvider bitbucketClientFactoryProvider;
+        @Inject
+        private BitbucketPluginConfiguration bitbucketPluginConfiguration;
+        @Inject
+        private BitbucketScmFormFillDelegate formFill;
+        @Inject
+        private BitbucketScmFormValidationDelegate formValidation;
+        @Inject
+        private JenkinsToBitbucketCredentials jenkinsToBitbucketCredentials;
```

Comment:
supernit: Interface injection into the descriptor will cause the plugin to crash in version 2.212 - 2.216. You can get around this with a provider, as in https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/108. Unsure if it's worth the effort to do or not.

Reply:
Is it worth dealing with? I don't know, I can't judge if it is a case we need to cover? I'll take your advice on this.